src/Invoker.php

```diff
@@ -30,19 +31,33 @@ class Invoker
     private $errorLogFunc;
 
     /**
-     * @param callable $target      The callable to be invoked
-     * @param string $signatureType The signature type of the target callable,
-     *                              either "event" or "http".
+     * @param array|string $target      The callable to be invoked
      */
-    public function __construct(callable $target, string $signatureType)
+    public function __construct($target)
     {
+        $signatureType = getenv('FUNCTION_SIGNATURE_TYPE', true) ?: 'http';
```

Comment:
It was an intentional choice to have the environment variable passed in. Injecting the environment variables is a better practice, easier to test, more reliable than having the logic exist in the `Invoker` itself. Also, this breaks backwards compatibility, if users were calling the `Invoker` directly. I think we need to roll back this change.

Reply:
The issue I ran into was that I couldn't test CloudEvent functions properly with this code in `router.php`, because it is just a script and I couldn't figure out how to pass it a CloudEvent request.

I can add back the signature type parameter here, and then only read the env var if it's missing. That way it is not a change to the constructor?

Can we mark these classes as internal? In general, I don't think we want people to be using these directly.